routing/routing_helpers.hpp

```diff
@@ -30,4 +32,41 @@ bool IsRoad(TTypes const & types)
 void ReconstructRoute(IDirectionsEngine & engine, RoadGraphBase const & graph,
                       shared_ptr<TrafficStash> const & trafficStash,
                       my::Cancellable const & cancellable, vector<Junction> & path, Route & route);
+
+/// \brief Checks is edge connected with world graph. Function does BFS while it finds some number of edges,
+/// if graph ends before this number is reached then junction is assumed as not connected to the world graph.
+template <typename GraphEdge, typename Graph, typename GraphVertex, typename GetVertexByEdgeFn, typename GetOutgoingEdgesFn>
+bool CheckGraphConnectivity(GraphVertex const & vertex,
+                            size_t limit,
+                            Graph & graph,
```

Comment:
Why not to use Vertex and Edge from Graph class? I mean, is it possible to do following:
```c++
using Vertex = typename Graph::Vertex;
using Edge = typename Graph::Edge;
...
```
?